server/ctrladmin/handlers_raw.go

```diff
@@ -18,10 +19,18 @@ func (c *Controller) ServeLoginDo(w http.ResponseWriter, r *http.Request) {
 	}
 	user := c.dbc.GetUserByName(username)
 	if user == nil || password != user.Password {
-		sessAddFlashW(session, []string{"invalid username / password"})
-		sessLogSave(session, w, r)
-		http.Redirect(w, r, r.Referer(), http.StatusSeeOther)
-		return
+		// Because internal authentication failed, we can now try to use LDAP, if
```

Comment:
same as other comment (ill just paste it since i can't link it think)

-------

not sure i like the idea of "if user/pass fails, try LDAP" 

seems like it would be better to have

```
if ldap is enabled {
   do ldap
   return
}
do normal user name password stuff
```

this would require that LDAP is configured from the very start (see other comment)

Reply:
Fixed in 6ca68b2c260d855b8965b762621efd9396c16107.

Reply:
Finished in f5195f15c4042357417d866af66ce8fd9fd711ee.